### ISSUE TITLE
[Feature Request] Support `dtype` arg in `torch._foreach_norm`

### ISSUE THREAD

**Issue by awgu** [pytorch/pytorch#125040]
`torch.linalg.vector_norm()` and `torch.norm` support the `dtype` argument. Can we get support for it in `torch._foreach_norm`?

cc @crcrpar @mcarilli @janeyx99

**Comment by crcrpar** [comment]
it should be doable.
just to confirm, with `dtype` we specify the dtype of output tensors, right?

ref: https://pytorch.org/docs/stable/generated/torch.norm.html

**Comment by awgu** [comment]
I think semantically, it should equivalent to casting inputs to `dtype` first:
<img width="810" alt="Screenshot 2024-04-26 at 12 45 37 PM" src="https://github.com/pytorch/pytorch/assets/31054793/4acc3aab-d0d7-482b-9298-02801b336f7e">
https://pytorch.org/docs/stable/generated/torch.linalg.vector_norm.html

I am mostly interested in this behavior. It is not a high priority though because in most cases, the tensors are already fp32 (e.g. gradients for gradient norm clipping).

**Comment by ad8e** [comment]
I think this issue is not closed because the linked merge was closed by hand rather than by the bot.
